projects/igniteui-angular/src/lib/grids/common/enums.ts

```diff
@@ -29,35 +54,61 @@ export type GridKeydownTargetType =
     'headerCell' |
     'masterDetailRow';
 
+/**
+ * Enumeration representing different selection modes for the grid columns if can be selected.
+ * - none: No selection is allowed.
+ */
 export const GridSelectionMode = mkenum({
     none: 'none',
     single: 'single',
     multiple: 'multiple',
     multipleCascade: 'multipleCascade'
 });
+/** Type alias for the keys of the GridSelectionMode enumeration. */
 export type GridSelectionMode = (typeof GridSelectionMode)[keyof typeof GridSelectionMode];
 
+/** Enumeration representing different column display order options. */
 export const ColumnDisplayOrder = mkenum({
     Alphabetical: 'Alphabetical',
     DisplayOrder: 'DisplayOrder'
 });
+/** Type alias for the keys of the ColumnDisplayOrder enumeration. */
 export type ColumnDisplayOrder = (typeof ColumnDisplayOrder)[keyof typeof ColumnDisplayOrder];
 
+/**
+ * Enumeration representing the possible positions for pinning columns.
+ * - Start: Columns are pinned to the start of the grid.
+ * - End: Columns are pinned to the end of the grid.
+ */
 export enum ColumnPinningPosition {
     Start,
     End
 }
 
+/**
+ * Enumeration representing the possible positions for pinning rows.
+ * - Top: Rows are pinned to the top of the grid.
+ * - Bottom: Rows are pinned to the bottom of the grid.
+ */
 export enum RowPinningPosition {
     Top,
     Bottom
 }
 
+/**
+ * Enumeration representing different paging modes for the grid.
+ * - Local: Default value; The grid will paginate the data source based on the page
```

Comment:
```suggestion
 * - Local: The grid will use local data to extract pages during paging.
 * - Remote: The grid will expect pages to be delivered from a remote location and will only raise events during paging interactions.
```

Reply:
Will include in the commit

Reply:
@GrblackStar Just FYI, for the suggested comments, you can click "commit suggestion" from the UI. You do not have to replicate it in a different commit.